core/extrinsics_submission_service/impl/service_builder.hpp

```diff
@@ -0,0 +1,31 @@
+/**
+ * Copyright Soramitsu Co., Ltd. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#ifndef KAGOME_CORE_EXTRINSICS_SUBMISSION_SERVICE_SERVICE_BUILDER_HPP
+#define KAGOME_CORE_EXTRINSICS_SUBMISSION_SERVICE_SERVICE_BUILDER_HPP
+
+#include <boost/asio/io_context.hpp>
+
+#include "extrinsics_submission_service/extrinsic_submission_proxy.hpp"
+#include "extrinsics_submission_service/json_transport.hpp"
+#include "extrinsics_submission_service/service.hpp"
+#include "extrinsics_submission_service/service_configuration.hpp"
+
+namespace kagome::service {
+  class ExtrinsicSubmissionServiceBuilder {
+    template <class T>
+    using sptr = std::shared_ptr<T>;
+
+   public:
+    std::shared_ptr<ExtrinsicSubmissionService> build(
```

Comment:
I don't think builder is necessary here.

if it was like
```
set_context(...)

set_configuatioin(...)
...
build()
```
then it would be builder. 
this class does not look like necessary. 

you could do this "build" thing in constructor of ExtrinsicSubmissionService

Reply:
what if I want to create a test version of service? in this case I would make a TestServiceBuilder class, composing a service from stubs.

Reply:
you always can do the same via ExtrinsicSubmissionService constructor

Reply:
removed builde